Add React Core Package

0xcadams opened the pull request:
## Description

Add react core package.

jozanza commented on packages/graphql-sdk/src/index.test.js:
```diff
@@ -18,6 +18,7 @@ import {
   TranscoderQuery,
   TranscodersQuery,
   ProtocolQuery,
+  ProtocolsQuery,
```
@mkpdev I can't think of a reason we need a "ProtocolsQuery" since this is just overview information about the Livepeer protocol itself. There is only 1 protocol, so we should remove any code related to this query.

jozanza commented on packages/graphql-sdk/src/queries.js:
```diff
@@ -156,8 +156,16 @@ query TranscodersQuery($skip: Int, $limit: Int) {
 export const ProtocolQuery = `
 ${ProtocolFragment}
-query ProtocolQuery {
-  protocol {
```
You can revert the changes in this file. The protocol query does not need an id as a parameter, and there should not be any sort of `ProtocolsQuery`

jozanza commented on packages/graphql-sdk/src/types/Query.js:
```diff
@@ -57,7 +57,10 @@ type Query {
   transcoders(skip: Int, limit: Int): [Transcoder!]!
 
   "The protocol as a whole"
-  protocol: Protocol!
```
we should revert this as well.

jozanza commented on packages/graphql-sdk/src/resolvers/Query.js:
```diff
@@ -322,16 +328,31 @@ export async function transcoders(
-  }
+): Protocol {
+  const { id } = args
+  const protocol = await ctx.livepeer.rpc.getProtocol(id)
```
@mkpdev There is no `getProtocol()` method in the sdk currently. You will have to add this one yourself (and it shouldn't need any arguments).

jozanza commented on packages/graphql-sdk/src/resolvers/Query.js:
```diff
@@ -322,16 +328,31 @@ export async function transcoders(
+ * @param {GQLContext} ctx
+ * @return {Protocols}
+ */
+export async function protocols(
```
We can remove this resolver

jozanza commented on packages/graphql-sdk/src/resolvers/Protocol.js:
```diff
@@ -16,7 +16,7 @@ type ProtocolObj = {}
  * @return {string}
  */
 export function id(obj: ProtocolObj): string {
-  return 'protocol'
```
This can remain hard-coded to "protocol"; This field is only really required since Apollo relies on the `id` field for caching.

jozanza commented on packages/graphql-sdk/src/queries.js:
```diff
@@ -156,8 +156,16 @@ query TranscodersQuery($skip: Int, $limit: Int) {
+  }
+}
+`
+export const ProtocolsQuery = `
```
we can remove this query

jozanza commented on packages/graphql-sdk/src/mock-sdk.js:
```diff
@@ -222,4 +223,23 @@ const ALL_TRANSCODERS = {
   },
 }
 
+const ALL_PROTOCOLS = [
```
We can remove  `ALL_PROTOCOLS` here -- just one `PROTOCOL` object.

jozanza commented on packages/graphql-sdk/src/mock-sdk.js:
```diff
@@ -64,7 +64,8 @@ const livepeer = {
     getTranscoders: async () => Object.values(ALL_TRANSCODERS),
-    getProtocolPaused: async () => false,
+    getProtocol: async id => ALL_PROTOCOLS[id],
+    getProtocols: async () => ALL_PROTOCOLS,
```
We can remove this

jozanza commented on packages/graphql-sdk/src/index.test.js:
```diff
@@ -362,7 +363,16 @@ test('TranscodersQuery', async t => {
  */
 test('ProtocolQuery', async t => {
-  const res = await graphql(schema, ProtocolQuery, null, { livepeer }, {})
-  // console.log(JSON.stringify(res, null, 2))
```
We can revert this test (no args required here)

jozanza commented on packages/graphql-sdk/src/index.test.js:
```diff
@@ -362,7 +363,16 @@ test('TranscodersQuery', async t => {
+  t.snapshot(res)
+})
+
+test('ProtocolsQuery', async t => {
```
Since we don't need to query for multiple Protocols (there is only one singleton), we can kill this test

jozanza commented:
Oh and @mkpdev please update your initial comment on this thread to reflect to format specified in https://raw.githubusercontent.com/livepeer/livepeerjs/master/PULL_REQUEST_TEMPLATE.md

mkpdev commented:
okay @jozanza will do mentioned changes. Thanks!

ericxtang commented on packages/graphql-sdk/schema.json:
```diff
@@ -423,14 +423,49 @@
-                  "kind": "OBJECT",
-                  "name": "Protocol",
-                  "ofType": null
+                  "kind": "LIST",
```
@mkpdev I think in this case, the protocol is a "singleton", since we will only have a single protocol to refer to.  For example, we won't have to look up a protocol by its ID.

mkpdev replied:
@ericxtang I fixed it. Please check.

Randy1Burrell commented on packages/graphql-sdk/src/resolvers/Query.js:
```diff
@@ -322,16 +323,14 @@ export async function transcoders(
 export async function protocol(
   obj: QueryObj,
-  args: QueryCurrentRoundArgs,
+  args: QueryProtocolArgs,
```
@mkpdev Thanks for a great work. Quick question about this part of the code. It seems to be giving an code quality error, and I have been looking at it a little. Just wondering, why are these params (obj and args) needed?

mkpdev replied:
@Randy1Burrell This function was already written I just change args from `QueryCurrentRoundArgs` to `QueryProtocolArgs` both have blank hash `{}`.

Also each function of this file accept 3 arguments `obj, args, ctx`.

ericxtang commented on packages/sdk/src/index.js:
```diff
@@ -549,7 +549,7 @@ export default async function createLivepeerSDK(
 ): Promise<LivepeerSDK> {
   const { ens, events, ...config } = await initContracts(opts)
-  const {
+  const {>>>>>>> master
```
@Randy1Burrell looks like a typo here.

Randy1Burrell replied:
Hmmm, seems this happened because I didn't remove the git conflict tags. I will schedule time to deal with these soon.

Randy1Burrell commented:
@mkpdev Thanks for your contribution.

0xcadams committed:
Merge commit 'c61dff7fcaa58ebeb2c00c0cc03934a7fe7a894d' into 0xcadams/access-control

0xcadams committed:
Merge remote-tracking branch 'origin/main' into 0xcadams/access-control

0xcadams committed:
docs: update nextra dependency

0xcadams committed:
feat: added crypto package and work on signed playback

0xcadams committed:
Merge remote-tracking branch 'origin/main' into 0xcadams/access-control

0xcadams committed:
Merge remote-tracking branch 'origin/main' into 0xcadams/access-control

0xcadams committed:
fix: update deps

0xcadams committed:
fix: types

0xcadams committed:
Merge branch 'main' into 0xcadams/access-control

0xcadams committed:
fix: move to using rainbowkit

0xcadams committed:
feat: removed dependencies and used webcrypto for jwt signing

0xcadams committed:
chore: add initial expo dev example app

0xcadams committed:
Merge branch '0xcadams/access-control' into 0xcadams/rn

0xcadams committed:
chore: added react native bare dev

0xcadams committed:
fix: further work towards rn examples

Commit:
Added z-index to the default variant of the player (#132)

0xcadams committed:
Add nodejs example (#131)

* feat: added import script for nodejs example

* fix: small fixes and added comments

* fix: added handling of paths in dstorage

* fix: moved ipfs-playback to dstorage and added redirect

* fix: added output json and fix docs links

* fix: fix docs on contributing

dependabot[bot] committed:
chore(deps): bump pnpm/action-setup from 2.2.2 to 2.2.4 (#125)

Bumps [pnpm/action-setup](https://github.com/pnpm/action-setup) from 2.2.2 to 2.2.4.
- [Release notes](https://github.com/pnpm/action-setup/releases)
- [Commits](https://github.com/pnpm/action-setup/compare/v2.2.2...v2.2.4)

---
updated-dependencies:
- dependency-name: pnpm/action-setup
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

Commit:
Added changeset for z-index change (#135)

0xcadams committed:
fix: added HEAD request to check for valid URLs

0xcadams committed:
Merge remote-tracking branch 'origin/main'

0xcadams committed:
Add access control docs (#93)

* docs: update nextra dependency

* feat: added crypto package and work on signed playback

* fix: update deps

* fix: types

* fix: move to using rainbowkit

* feat: removed dependencies and used webcrypto for jwt signing

* feat: added further access control

* fix: added b64 decoding for private key

* fix: final fixes for access control

* fix: fixed packages

* docs: clean up svelte example

* chore: fix up issue templates

suhailkakar committed:
docs: removed extra brackets from useCreateAsset example (#136)

Co-authored-by: Chase Adams <c@cadams.io>

pglowacky committed:
update docs (#138)

* update docs

* Update docs/pages/examples/react/dstorage-playback.en-US.mdx

Co-authored-by: Chase Adams <c@cadams.io>

Co-authored-by: Chase Adams <c@cadams.io>

github-actions[bot] committed:
chore: version packages (#134)

* chore: version packages

* ci: bump

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Chase Adams <c@cadams.io>

suhailkakar committed:
Add support for multiple files at once using useCreateAsset  (#102)

* feat: added multi-video upload to useCreateAsset hook

* fix: updated changeset description

* fix: solved the Asset.tsx in example component error

* feat: added multiple file upload at the core package

* fix: fixed the build error in next example directory

* fix: refactor types

* fix: updated the changeset and types

* docs: updated the types in useCreateAsset page

* fix: fixed the merge conflicts

* docs: updated the types in the documentation.

* refactor: removed styles and console.log's

* refactor: replaced promise.all with promise.allSettled

* docs: updated the types in the docs

* refactor: updated the upload progress object

* fix: reverted the types

- refactor: updated the return in Promise.allSettled

* fix: fixed the typed lint issue

* refactor: remove the uploadsize from tus

* fix: fix types

* refactor: removed meta from the tus upload

* fix: refactored types

* docs: added new types to the docs

Co-authored-by: Chase Adams <c@cadams.io>

0xcadams committed:
Rework types to support multiple createAsset as const (#140)

* feat: added multi-video upload to useCreateAsset hook

* fix: updated changeset description

* fix: solved the Asset.tsx in example component error

* feat: added multiple file upload at the core package

* fix: fixed the build error in next example directory

* fix: refactor types

* fix: updated the changeset and types

* docs: updated the types in useCreateAsset page

* fix: fixed the merge conflicts

* docs: updated the types in the documentation.

* refactor: removed styles and console.log's

* refactor: replaced promise.all with promise.allSettled

* docs: updated the types in the docs

* refactor: updated the upload progress object

* fix: reverted the types

- refactor: updated the return in Promise.allSettled

* fix: fixed the typed lint issue

* refactor: remove the uploadsize from tus

* fix: fix types

* refactor: removed meta from the tus upload

* fix: refactored types

* docs: added new types to the docs

* fix: rework types to support const assertions for typing

* test: fix tests

Co-authored-by: Suhail Kakar <suhail.zahed@gmail.com>

0xcadams committed:
Merge remote-tracking branch 'origin/main' into 0xcadams/rn

0xcadams committed:
fix: merge cleanup

0xcadams committed:
Merge remote-tracking branch 'origin/main' into 0xcadams/rn

0xcadams committed:
test: fix tests

0xcadams committed:
fix: update nodejs example

0xcadams committed:
refactor: added core-react package and moved web logic out of react

0xcadams committed:
Merge remote-tracking branch 'origin/main' into 0xcadams/rn

0xcadams committed:
chore: changesets

0xcadams committed:
Revert "chore: changesets"

This reverts commit 498cfca9d1ff56a5dc3a9cfcfb702ed3ec8e1569.

0xcadams committed:
chore: changeset

0xcadams committed:
ci: add release on next branch

0xcadams committed:
chore: add pre

0xcadams committed:
fix: fix prerendering

0xcadams committed:
Merge remote-tracking branch 'origin/HEAD' into 0xcadams/rn

0xcadams committed:
style: lint

0xcadams committed:
fix: added core js to react

0xcadams committed:
test: fix tests

0xcadams committed:
test: fix

0xcadams committed:
Merge remote-tracking branch 'origin/HEAD' into 0xcadams/rn
